protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -53,102 +53,124 @@ func (k *Keeper) InitializeEquityTierLimit(
-
-	var equityTierLimits []types.EquityTierLimit
-	if order.IsShortTermOrder() {
-		equityTierLimits = k.GetEquityTierLimitConfiguration(ctx).ShortTermOrderEquityTiers
-	} else if order.IsStatefulOrder() {
-		equityTierLimits = k.GetEquityTierLimitConfiguration(ctx).StatefulOrderEquityTiers
-	} else {
-		panic(fmt.Sprintf("Unsupported order type for equity tiers. Order: %+v", order))
-	}
-	if len(equityTierLimits) == 0 {
-		return nil
-	}
-
-	subaccountId := order.GetSubaccountId()
+// getEquityTierLimit returns the equity tier limit for a subaccount based on its net collateral.
+func (k Keeper) getEquityTierLimit(
+	ctx sdk.Context, subaccountId satypes.SubaccountId,
+	equityTierLimits []types.EquityTierLimit,
+) (types.EquityTierLimit, error) {
 	netCollateral, _, _, err := k.subaccountsKeeper.GetNetCollateralAndMarginRequirements(
 		ctx,
 		satypes.Update{
 			SubaccountId: subaccountId,
 		},
 	)
 	if err != nil {
-		return err
+		return types.EquityTierLimit{}, err
 	}
 
-	equityTierLimit := types.EquityTierLimit{}
+	var equityTierLimit types.EquityTierLimit
 	for _, limit := range equityTierLimits {
 		if netCollateral.Cmp(limit.UsdTncRequired.BigInt()) < 0 {
 			break
 		}
 		equityTierLimit = limit
 	}
-	// Return immediately if the amount the subaccount can open is 0.
+
+	// Return immediately if the amount the subaccount can open is 0
 	if equityTierLimit.Limit == 0 {
+		return types.EquityTierLimit{}, errorsmod.Wrapf(
+			types.ErrOrderWouldExceedMaxOpenOrdersEquityTierLimit,
+			"Opening order would exceed equity tier limit of %d, for subaccount %+v with net collateral %+v",
+			equityTierLimit.Limit,
+			subaccountId,
+			netCollateral,
+		)
+	}
+
+	return equityTierLimit, nil
+}
+
+// ValidateSubaccountEquityTierLimitForShortTermOrder returns an error if adding the order would exceed the equity
+// tier limit on how many short term open orders a subaccount can have. Short-term fill-or-kill and immediate-or-cancel
+// orders never rest on the book and will always be allowed as they do not apply to the number of open orders that
+// equity tier limits enforce.
+func (k Keeper) ValidateSubaccountEquityTierLimitForShortTermOrder(ctx sdk.Context, order types.Order) error {
+	if order.RequiresImmediateExecution() {
+		return nil
+	}
+
+	equityTierLimits := k.GetEquityTierLimitConfiguration(ctx).ShortTermOrderEquityTiers
+	if len(equityTierLimits) == 0 {
+		return nil
+	}
+
+	equityTierLimit, err := k.getEquityTierLimit(ctx, order.GetSubaccountId(), equityTierLimits)
+	if err != nil {
+		return err
+	}
+
+	// For short term orders we just count how many orders exist on the memclob.
+	equityTierCount := k.MemClob.CountSubaccountShortTermOrders(ctx, order.GetSubaccountId())
+
+	// Verify that opening this order would not exceed the maximum amount of orders for the equity tier.
+	if lib.MustConvertIntegerToUint32(equityTierCount) >= equityTierLimit.Limit {
 		return errorsmod.Wrapf(
 			types.ErrOrderWouldExceedMaxOpenOrdersEquityTierLimit,
```

Comment:
Knowing the TNC was nice to have as part of this error.

Reply:
done